src/test/java/app/controller/impl/ReviewControllerImplTest.java

```diff
@@ -0,0 +1,193 @@
+    static void beforeAll() {
+        EntityManagerFactory emf = HibernateConfig.getEntityManagerFactoryForTest();
+        int port = 7070;
+
+        populatorTestUtil = new PopulatorTestUtil(emf);
+        app = AppConfig.startServer(port, emf);
+
+        gameMapper = new GameMapper();
+        reviewMapper = new ReviewMapper();
+
+        RestAssured.baseURI = String.format("http://localhost:%d/api", port);
+    }
+
+    @BeforeEach
+    void setUp() {
+        List<Role> roles = populatorTestUtil.createRoles();
+        populatorTestUtil.persist(roles);
+
+        List<User> users = populatorTestUtil.createUsers(roles);
+        populatorTestUtil.persist(users);
+
+        List<Genre> genres = populatorTestUtil.createGenres();
+        populatorTestUtil.persist(genres);
+
+        List<Platform> platforms = populatorTestUtil.createPlatforms();
+        populatorTestUtil.persist(platforms);
+
+        List<Game> games = populatorTestUtil.createGames(genres, platforms);
+        populatorTestUtil.persist(games);
+
+        List<Review> reviews = populatorTestUtil.createReviews(users, games);
+        populatorTestUtil.persist(reviews);
+
+        gameDTOList = games.stream().map(gameMapper::convertToDTO).toList();
+        reviewDTOList = reviews.stream().map(reviewMapper::convertToDTO).toList();
+    }
+
+    @AfterEach
+    void tearDown() {
+        populatorTestUtil.cleanup(User.class);
+        populatorTestUtil.cleanup(Role.class);
+
+        populatorTestUtil.cleanup(Review.class);
+        populatorTestUtil.cleanup(Game.class);
+        populatorTestUtil.cleanup(Genre.class);
+        populatorTestUtil.cleanup(Platform.class);
+    }
+
+    @AfterAll
+    static void afterAll() {
+        AppConfig.stopServer(app);
+    }
+
+    @Test
+    void getById() {
+        ReviewDTO expectedReviewDTO = reviewDTOList.get(2);
+
+        ReviewDTO actualReviewDTO = given()
+                .when()
+                .get("reviews/{id}", expectedReviewDTO.getId())
+                .then()
+                .statusCode(HttpStatus.OK.getCode())
+                .extract()
+                .as(ReviewDTO.class);
+
+        assertThat(actualReviewDTO, is(expectedReviewDTO));
+    }
+
+    @Test
+    void getAll() {
+        GameDTO gameDTO = gameDTOList.get(1);
+        List<ReviewDTO> expectedReviews = reviewDTOList.stream().filter(review -> review.getGameId().equals(gameDTO.getId())).toList();
+
+        List<ReviewDTO> actualReviews = given()
+                .when()
+                .get("/games/{id}/reviews", gameDTO.getId())
+                .then()
+                .statusCode(HttpStatus.OK.getCode())
+                .extract()
+                .jsonPath()
+                .getList("$", ReviewDTO.class);
+
+        assertThat(actualReviews.size(), is(expectedReviews.size()));
+        assertThat(actualReviews, containsInAnyOrder(expectedReviews.toArray()));
+    }
+
+    @Test
+    void create() {
+        String username = "User2";
```

Comment:
Should a UserDTO be instantiated with a username and a password and then it's getters can be used? (Not important)

Reply:
Decided not to make a user object to only get a username